Environment_Engine/Create/Panel.cs

```diff
@@ -40,6 +43,52 @@ public static partial class Create
+        [Input("brep", "A BHoM Boundary Representation to convert into a collection of Environment Panels")]
+        [Input("connectedSpaceName", "A name for the space which these panels are connected to. If no name is provided, a randomised default will be generated")]
+        [Input("angleTolerance", "The angle tolerance for collapsing to polylines used when generating the external edges of the surfaces")]
+        [Output("panelsAsSpace", "A collection of Environment Panels representing a closed space generated from the provided Brep geometry")]
+        public static List<Panel> Panels(this BoundaryRepresentation brep, string connectedSpaceName = null, double angleTolerance = BH.oM.Geometry.Tolerance.Angle)
+        {
+            return brep.Surfaces.ToList().Panels(connectedSpaceName, angleTolerance);
+        }
+
+        [Description("Create a collection of Environment Panels from a collection of BHoM Surfaces")]
+        [Input("surfaces", "A collection of BHoM surfaces to convert into a Environment Panels. The surfaces should be grouped as a single space as all panels generated from the surfaces will have the same connectedSpaceName")]
+        [Input("connectedSpaceName", "A name for the space which these panels are connected to. If no name is provided, a randomised default will be generated")]
+        [Input("angleTolerance", "The angle tolerance for collapsing to polylines used when generating the external edges of the surfaces")]
+        [Output("panel", "An Environment Panels representing a closed space generated from the provided surfaces")]
+        public static List<Panel> Panels(this List<ISurface> surfaces, string connectedSpaceName = null, double angleTolerance = BH.oM.Geometry.Tolerance.Angle)
+        {
+            if (connectedSpaceName == null)
+                connectedSpaceName = Guid.NewGuid().ToString();
+
+            List<Panel> panels = surfaces.Select(x => x.Panel(connectedSpaceName, angleTolerance)).ToList();
+
+            panels = panels.SetRoofPanels();
+            panels = panels.SetFloorPanels();
+            panels = panels.SetWallPanels();
+
+            return panels;
+        }
+
+        [Description("Create an Environments Panel from a BHoM Surface")]
+        [Input("surface", "A BHoM surface to convert into an Environment Panel")]
+        [Input("connectedSpaceName", "A name for the space which this panel is connected to. If no name is provided, a randomised default will be generated")]
+        [Input("angleTolerance", "The angle tolerance for collapsing to polylines used when generating the external edges of the surfaces")]
+        [Output("panel", "An Environment Panels representing a closed space generated from the provided Brep geometry")]
+        public static Panel Panel(this ISurface surface, string connectedSpaceName = null, double angleTolerance = BH.oM.Geometry.Tolerance.Angle)
```

Comment:
One more comment here, is that for at least PlanarSurfaces, good if the openings of it are turned into openings of the panel as well? That is at least what we have been doing for structures (we only have creates for planar surface, not for a generic surface)

Reply:
Cheers for the suggestion @IsakNaslundBh , have implemented using `IInternalEdges` from the surface.